[ISSUE] databricks_permissions fails with Provider produced inconsistent result when provided upgraded to 0.3.0

I was running below code with Databricks provider `0.2.4` and it was working fine. I am currently looking to upgrade provider to latest version `0.3.0` and terraform apply is failing with below error.

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

```
terraform -v
Terraform v0.12.21
```

### Affected Resource(s)
Please list the resources as a list, for example:
- databricks_permissions

### Environment variable names
To get relevant environment variable _names_ please copypaste the output of the following command:
`$ env | sort | grep -E 'DATABRICKS|AWS|AZURE|ARM|TEST' | awk -F= '{print $1}'`

```
ARM_ACCESS_KEY
ARM_CLIENT_ID
ARM_CLIENT_SECRET
ARM_SUBSCRIPTION_ID
ARM_TENANT_ID
```

### Terraform Configuration Files
```hcl
resource "databricks_cluster_policy" "vm_policy" {
  name       = "db_vm_cluster_policy"
  definition = jsonencode(merge(local.default_policy))
}

resource "databricks_permissions" "can_use_cluster_policy_vm_type" {
  cluster_policy_id = databricks_cluster_policy.vm_policy.id

  access_control {
    group_name = "admins"
    permission_level = "CAN_USE"
  }
}
```

### Debug Output
```
TF_LOG=DEBUG terraform plan 2>&1 | grep databricks | sed -E 's/^.* plugin[^:]+: (.*)$/\1/'

databricks_permissions.can_use_cluster_policy_vm_type: Creating...
2021/02/04 15:45:00 [DEBUG] databricks_permissions.can_use_cluster_policy_vm_type: applying the planned Create change
Using Azure Service Principal client secret authentication: timestamp=2021-02-04T15:45:00.156-0500
Getting Workspace ID via management token.: timestamp=2021-02-04T15:45:00.157-0500
GET https://management.azure.com/subscriptions/xxxx-xxx-xxxxxx-xxxxx/resourceGroups/xxxx-xxx-xxx/providers/Microsoft.Databricks/workspaces/xxxx-xxx-xxx-adb-xxx?api-version=2018-04-01: timestamp=2021-02-04T15:45:00.708-0500
200 OK {
    "managedResourceGroupId": "/subscriptions/xxxx-xxx-xxxxxx-xxxxx/resourceGroups/databricks-rg-xxxx-xxx-xxx",
        "value": "databricks-private"
        "value": "databricks-public"
Creating workspace token: timestamp=2021-02-04T15:45:01.450-0500
Setting 'X-Databricks-Azure-SP-Management-Token' header: timestamp=2021-02-04T15:45:01.450-0500
POST https://adb-XXXXXXXX.azuredatabricks.net/api/2.0/token/create {
200 OK {
} <- POST https://adb-XXXXXXXX.azuredatabricks.net/api/2.0/token/create: timestamp=2021-02-04T15:45:02.939-0500
PUT /preview/permissions/cluster-policies/XXXXXXXX {
200 OK {
GET /preview/permissions/cluster-policies/XXXXXXXX: timestamp=2021-02-04T15:45:03.427-0500
200 OK {
GET /preview/scim/v2/Me: timestamp=2021-02-04T15:45:03.729-0500
200 OK {
2021/02/04 15:45:04 [DEBUG] databricks_permissions.can_use_cluster_policy_vm_type: apply errored, 
            but we're indicating that via the Error pointer rather than returning it: 
            Provider produced inconsistent result after apply: When applying changes to 
            databricks_permissions.can_use_cluster_policy_vm_type, 
            provider "registry.terraform.io/-/databricks" produced an unexpected new value for was present, but now absent.
2021/02/04 15:45:04 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: Provider produced inconsistent result after apply: 
            When applying changes to databricks_permissions.can_use_cluster_policy_vm_type, provider "registry.terraform.io/-/databricks" 
            produced an unexpected new value for was present, but now absent.
2021/02/04 15:45:04 [ERROR] <root>: eval: *terraform.EvalSequence, err: Provider produced inconsistent result after apply: 
            When applying changes to databricks_permissions.can_use_cluster_policy_vm_type, provider "registry.terraform.io/-/databricks" 
            produced an unexpected new value for was present, but now absent.
databricks_permissions.can_use_cluster_policy_vm_type, provider
"registry.terraform.io/-/databricks" produced an unexpected new value for was
path=/Users/npatel/.terraform.d/plugins/darwin_amd64/terraform-provider-databricks_v0.3.0 pid=50166
```

### Expected Behavior
I think, it should be able to assign policy to **admins** group. (I am not sure if Databricks doesn't allow to update admins group)

### Actual Behavior
```
databricks_permissions.can_use_cluster_policy_vm_type: Creating...

Error: Provider produced inconsistent result after apply

When applying changes to
databricks_permissions.can_use_cluster_policy_vm_type, provider
"registry.terraform.io/-/databricks" produced an unexpected new value for was
present, but now absent.
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Are there anything atypical about your accounts that we should know?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Version changelog
 
+## 0.3.1
+
+**Behavior changes**
+
+* Assigning any permission to `admins` would result in an error, so that behavior is consistent ([#486](https://github.com/databrickslabs/terraform-provider-databricks/issues/486)).
+
 ## 0.3.0
 
 * Added configurable provisioning timeout for `databricks_mws_workspaces`, so that local DNS cache issues would be more tolerated.

--- a/access/resource_permissions.go
+++ b/access/resource_permissions.go
@@ -12,6 +12,7 @@ import (
 	"github.com/databrickslabs/databricks-terraform/identity"
 	"github.com/databrickslabs/databricks-terraform/internal"
 	"github.com/databrickslabs/databricks-terraform/workspace"
+	"github.com/hashicorp/go-cty/cty"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/pkg/errors"
@@ -255,6 +256,23 @@ func ResourcePermissions() *schema.Resource {
 			}
 		}
 		s["access_control"].MinItems = 1
+		if groupNameSchema, err := internal.SchemaPath(s,
+			"access_control", "group_name"); err == nil {
+			groupNameSchema.ValidateDiagFunc = func(i interface{}, p cty.Path) diag.Diagnostics {
+				if v, ok := i.(string); ok {
+					if "admins" == strings.ToLower(v) {
+						return diag.Diagnostics{
+							{
+								Summary:       "It is not possible to restrict any permissions from `admins`.",
+								Severity:      diag.Error,
+								AttributePath: p,
+							},
+						}
+					}
+				}
+				return nil
+			}
+		}
 		return s
 	})
 	readContext := func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {

--- a/access/resource_permissions_test.go
+++ b/access/resource_permissions_test.go
@@ -299,6 +299,23 @@ func TestResourcePermissionsCreate_conflicting_fields(t *testing.T) {
 	qa.AssertErrorStartsWith(t, err, "Invalid config supplied. cluster_id: conflicts with notebook_path. notebook_path: conflicts with cluster_id")
 }
 
+func TestResourcePermissionsCreate_AdminsThrowError(t *testing.T) {
+	_, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{},
+		Resource: ResourcePermissions(),
+		Create:   true,
+		HCL: `
+		cluster_id = "abc"
+		access_control {
+			group_name = "admins"
+			permission_level = "CAN_MANAGE"
+		}
+		`,
+	}.Apply(t)
+	assert.EqualError(t, err, "Invalid config supplied. [access_control] "+
+		"It is not possible to restrict any permissions from `admins`.")
+}
+
 func TestResourcePermissionsCreate(t *testing.T) {
 	d, err := qa.ResourceFixture{
 		Fixtures: []qa.HTTPFixture{

--- a/go.mod
+++ b/go.mod
@@ -20,5 +20,6 @@ require (
 	github.com/stretchr/testify v1.7.0
 	github.com/zclconf/go-cty v1.2.1
 	golang.org/x/time v0.0.0-20191024005414-555d28b269f0
+	golang.org/x/tools v0.1.0 // indirect
 	gopkg.in/ini.v1 v1.62.0
 )

